.ci/lib/stage-test-simple-nosgx.jenkinsfile

```diff
@@ -0,0 +1,174 @@
+            then
+                USE_SELECT=1
+                export USE_SELECT
+            fi
+            make -j8 all
+            make start-graphene-server &
+            sleep 1
+            ./src/src/redis-benchmark
+        '''
+    }
+    timeout(time: 10, unit: 'MINUTES') {
+        sh '''
+            cd Examples/lighttpd
+            make -j8 all
+            make start-graphene-server &
+            sleep 1
+            LOOP=1 CONCURRENCY_LIST="1 32" ./benchmark-http.sh 127.0.0.1:8003
+        '''
+    }
+    timeout(time: 10, unit: 'MINUTES') {
+        sh '''
+            cd Examples/nginx
+            make -j8 all
+            make start-graphene-server &
+            sleep 1
+            LOOP=1 CONCURRENCY_LIST="1 32" ./benchmark-http.sh 127.0.0.1:8002
+        '''
+    }
+    timeout(time: 20, unit: 'MINUTES') {
+        sh '''
+            cd Examples/apache
+            make -j8 all
+            make start-graphene-server &
+            sleep 1
+            LOOP=1 CONCURRENCY_LIST="1 32" ./benchmark-http.sh 127.0.0.1:8001
+            LOOP=1 CONCURRENCY_LIST="1 32" ./benchmark-http.sh https://127.0.0.1:8443
+        '''
+    }
+    timeout(time: 5, unit: 'MINUTES') {
+        sh '''
+            cd Examples/blender
+            make ${MAKEOPTS} all
+            make check
+        '''
+    }
+    timeout(time: 5, unit: 'MINUTES') {
+        sh '''
+            cd Examples/r
+            make ${MAKEOPTS} all
+            make check
+        '''
+    }
+    sh '''
+        # Workaround LTP bug (see https://github.com/linux-test-project/ltp/issues/560 for upstream fix):
+        git -C LibOS/shim/test/ltp/src checkout -- utils/ffsb-6.0-rc2/config.h.in utils/ffsb-6.0-rc2/configure
+
+        ./Scripts/gitignore-test
+    '''
+    sh '''
+        cd "$(./Scripts/clean-check-test-copy)"
+
+        make distclean
+        if .ci/isdistro xenial
+        then
+            rm -r LibOS/glibc-2.27 LibOS/glibc-2.27.tar.gz
+        fi
+        make -C LibOS/shim/test/regression clean
+
+        make -C LibOS/shim/test/ltp clean
+        # LTP's make clean is broken, see https://github.com/linux-test-project/ltp/issues/559
+        rm -rf /tmp/graphene-*.clean-check.clean/LibOS/shim/test/ltp/src
+        rm -rf LibOS/shim/test/ltp/src
+
+        make -C Examples/python-simple clean
+        make -C Examples/python-scipy-insecure clean
+        make -C Examples/python-sh-insecure clean
+        make -C Examples/bash clean
+        make -C Examples/curl clean
+        make -C Examples/gcc distclean
+        make -C Examples/memcached distclean
+        make -C Examples/redis distclean
+        make -C Examples/lighttpd distclean
+        make -C Examples/nginx distclean
+        make -C Examples/apache distclean
+        make -C Examples/blender distclean
+        make -C Examples/r distclean
+        make -C Pal/src PAL_HOST=Skeleton clean
+
+        ./Scripts/clean-check
+    '''
+
+    load '.ci/lib/artifacts.jenkinsfile'
```

Comment:
Will the artifacts.jenkinsfile be run in case there were errors in the lines above? It's a always block where you want to copy the result XML files from the suite runs, so should it be separate post -> always block in the actual Jenkins file like the ci/linux-ubuntu16.04-gcc-nosgx-debug.jenkinsfile?

Reply:
Sounds good.